### PR TITLE
Testing multiple pods using label selector

### PR DESCRIPTION
**Pull request by knabben**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Adds a test for multiple pod selector with label

**Which issue(s) this PR fixes**:

Ref https://github.com/kubernetes/kubernetes/issues/97640

**Special notes for your reviewer**:
10 - Allow traffic multiple selector (OR)

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/network/netpol/network_policy.go [hunk 1]
```diff
@@ -350,6 +350,34 @@ var _ = SIGDescribeCopy("Netpol [LinuxOnly]", func() {
...
+					PodSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"pod": "c"},
+					},
+				},
+			}
+			policy := GetAllowIngressByMultiLabeledPod("allow-nsx-pod-b-c", map[string]string{"pod": "a"}, peers)
+			CreatePolicy(k8s, policy, nsX)
+
+			reachability := NewReachability(model.AllPods(), true)
+			for _, ns := range []string{nsX, nsY, nsZ} { // Disconnect all pods to Pod a on namespace nsX
```
Comment (mattfenwick) [inline test/e2e/network/netpol/network_policy.go:370]: This for-loop can be replaced with `reachability.ExpectAllIngress(NewPodString(nsX, "a"), false)`

File: test/e2e/network/netpol/policies.go [hunk 2]
```diff
@@ -244,6 +244,26 @@ func GetAllowIngressByNamespaceOrPod(name string, targetLabels map[string]string
 	return policy
 }
 
+// GetAllowIngressByMultiLabeledPod allows ingress for pods with matching multiple pod labels
+func GetAllowIngressByMultiLabeledPod(name string, targetLabels map[string]string, peers []networkingv1.NetworkPolicyPeer) *networkingv1.NetworkPolicy {
```
Comment (mattfenwick) [inline test/e2e/network/netpol/policies.go:248]: This name makes me think that the pod needs to have multiple labels in order to be matched -- whereas the intent is that the pod needs to have any of the labels in order to match, right?  Can the name be updated to clarify that?

File: test/e2e/network/netpol/network_policy.go [hunk 3]
```diff
@@ -350,6 +350,34 @@ var _ = SIGDescribeCopy("Netpol [LinuxOnly]", func() {
 			ValidateOrFail(k8s, model, &TestCase{FromPort: 81, ToPort: 80, Protocol: v1.ProtocolTCP, Reachability: reachability})
 		})
 
+		ginkgo.It("should enforce policy based on multiple label PodSelectors (OR) [Feature:NetworkPolicy]", func() {
+			nsX, nsY, nsZ, model, k8s := getK8SModel(f)
+			peers := []networkingv1.NetworkPolicyPeer{
```
Comment (mattfenwick) [inline test/e2e/network/netpol/network_policy.go:355]: wondering if these peers would be better off defined in the `GetAllowIngressByMultiLabeledPod` function

🤔 not sure

File: test/e2e/network/netpol/network_policy.go (hunk 1 above)
Comment (knabben) [reply test/e2e/network/netpol/network_policy.go:370]: good suggestions... I'm getting used to it!

File: test/e2e/network/netpol/policies.go (hunk 2 above)
Comment (knabben) [reply test/e2e/network/netpol/policies.go:248]: That's correct, the labels are treated as OR, maybe `GetAllowIngressByAnyPod`, this follows the semantics of using Pod for PodSelectors, like `// GetAllowIngressByPod allows ingress by pod labels`

File: test/e2e/network/netpol/network_policy.go (hunk 3 above)
Comment (knabben) [reply test/e2e/network/netpol/network_policy.go:355]: Maybe passing a list of dicts `[]map[string]string{ {"pod": "a", "pod": "b"} }`, and handling inside the function
```
func GetAllowIngressByAnyPod(name string, targetLabels map[string]string, peersLabel []map[string]string) *networkingv1.NetworkPolicy {
	policyPeers := []networkingv1.NetworkPolicyPeer{}
	for _, label := range peersLabel {
		policyPeers = append(policyPeers, networkingv1.NetworkPolicyPeer{
			PodSelector: &metav1.LabelSelector{MatchLabels: label},
		})
	}
```

File: test/e2e/network/netpol/network_policy.go (hunk 1 above)
Comment (knabben) [reply test/e2e/network/netpol/network_policy.go:370]: good suggestions!

**Pull request merged**